bouncer/src/repo/lib/repo_license.cpp

```diff
@@ -0,0 +1,207 @@
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "repo_license.h"
+
+namespace Licensing
+{
+
+#ifdef REPO_LICENSE_CHECK
+
+	// definitions for class static variables
+	std::string LicenseValidator::instanceUuid;
+	std::string LicenseValidator::license;
+	std::unique_ptr<Cryptolens> LicenseValidator::cryptolensHandle;
+
+	std::string LicenseValidator::GetInstanceUuid()
+	{
+		std::string instanceUuid = repo::lib::getEnvString(instanceUuidEnvVarName);
+		if (instanceUuid.empty())
+		{
+			instanceUuid = repo::lib::RepoUUID::createUUID().toString();
+		}
+		return instanceUuid;
+	}
+
+	std::string LicenseValidator::GetLicenseString()
+	{
+		std::string licenseStr = repo::lib::getEnvString(licenseEnvVarName);
+		if(licenseStr.empty())
+		{
+			std::stringstream ss;
+			ss << "License not found, expected to find it in this " <<
+				"environment variable: " <<
+				licenseEnvVarName;
+			throw repo::lib::RepoInvalidLicenseException(ss.str());
+		}
+		return licenseStr;
+	}
+
+	std::string LicenseValidator::GetFormattedUtcTime(time_t timeStamp)
+	{
+		struct tm  tstruct = *gmtime(&timeStamp);
+		char buf[80];
+		strftime(buf, sizeof(buf), "%Y/%m/%d %X", &tstruct);
+		std::string formatedStr(buf);
+		return formatedStr;
+	}
+
+	void LicenseValidator::Reset()
+	{
+		cryptolensHandle.reset();
+		instanceUuid.clear();
+	}
+
+#endif
+
+	void LicenseValidator::RunActivation()
+	{
+#ifdef REPO_LICENSE_CHECK
+		// only run once 
+		if (!instanceUuid.empty() || cryptolensHandle)
+		{
+			repoError << " Attempting to activate more than once, aborting activation";
+			return;
+		}
+
+		// Setting up the handle
+		license = GetLicenseString();
+		cryptolens::Error e;
+		cryptolensHandle = std::unique_ptr<Cryptolens>(new Cryptolens(e));
+		// setting the public key
+		cryptolensHandle->signature_verifier.set_modulus_base64(e, pubKeyModulus);
+		cryptolensHandle->signature_verifier.set_exponent_base64(e, pubKeyExponent);
+		// Using machine code to store the instance UUID instead, as per
+		// https://help.cryptolens.io/licensing-models/containers
+		instanceUuid = GetInstanceUuid();
+		cryptolensHandle->machine_code_computer.set_machine_code(e, instanceUuid);
+
+		// License activation 
+		cryptolens::optional<cryptolens::LicenseKey> licenseKey =
+			cryptolensHandle->activate_floating
+			(
+				e, // Object used for reporting if an error occured
+				authToken, // Cryptolens Access Token
+				productId, // Product key associated with the 3drepo.io pipeline
+				license, // License Key
+				floatingTimeIntervalSec // The amount of time the user has to wait before an actived machine (or session in our case) is taken off the license.
+			);
+
+		// dealing with early bail out scenarios
+		repoInfo << activationSummaryBlock;
+		if (e)
+		{
+			cryptolens::ActivateError error = cryptolens::ActivateError::from_reason(e.get_reason());
+			repoInfo << "- server message: " << error.what();
```

Comment:
can we change these to repoTrace?

And just have a simple one line pass/failed message in `repoInfo`. (Also do this for deactivation)

Whilst it's nice to know the details, we don't care on an operational basis and it's quite a bit of clutter in the log, the user can run in verbose mode to get more if required